src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ClusteringGenomicHMMSegmenter.java

```diff
@@ -19,31 +23,30 @@
  *
  * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
  */
-public abstract class ClusteringGenomicHMMSegmenter<T> {
+public abstract class ClusteringGenomicHMMSegmenter<DATA, HIDDEN> {
     protected final Logger logger = LogManager.getLogger(ClusteringGenomicHMMSegmenter.class);
 
     private double concentration;
     protected double[] weights; //one per hidden state
-    protected double[] hiddenStateValues;
-    protected double memoryLength;
+    protected List<HIDDEN> hiddenStateValues;
+    private double memoryLength;
     private boolean parametersHaveBeenLearned = false;
 
-    protected static final int NEUTRAL_VALUE_INDEX = 0;
-
     private static final int MAX_EM_ITERATIONS = 50;
-    protected final List<T> data;
+    protected final List<DATA> data;
+
     protected final List<SimpleInterval> positions;
     private final double[] distances;   // distances[n] is the n to n+1 distance
 
     protected static double NEGLIGIBLE_POSTERIOR_FOR_M_STEP = 0.001;
 
     private static final double MINIMUM_MEMORY_LENGTH = 1;
     private static final double MAXIMUM_MEMORY_LENGTH = 1e10;
-    private static final double DEFAULT_MEMORY_LENGTH = 5e7;
 
     // parameters for pruning unused hidden states
-    final double DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_SPURIOUS = 0.01;  // if a hidden state is this close to another state, it might be false
-    final double MAX_WEIGHT_CONSIDERED_FOR_PRUNING = 0.02;  // never prune a state with greater weight than this
+    final double DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_SPURIOUS = 0.02;  // if a hidden state is this close to another state, it might be false
```

Comment:
Since this will get a bit messy, do you mind if I do this after optimizing the segmentation code and model?  Pruning is going to be heavily affected.